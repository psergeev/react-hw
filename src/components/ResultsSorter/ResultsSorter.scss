.results-sorter {
    background-color: #F5F5F5;
    height: 40px;
    line-height: 40px;
    display: flex;
    justify-content: space-between;

    &>div {
        margin: 0 50px;
        color: #666;
    }
}
