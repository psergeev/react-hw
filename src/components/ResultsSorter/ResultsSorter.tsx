import * as React from 'react';
import SortByButton from '../SortByButton';
import './ResultsSorter.scss';

interface Props {
    moviesCount: number;
    sortBy: string;
    onSortByChange: Function;
}

export default React.memo((props: Props) => (
    <div className="results-sorter">
        <div className={!props.moviesCount ? 'hidden' : ''}>
            <div>{`${props.moviesCount} movie(s) found`}</div>
            <div>
                <span>Sort by</span>
                <SortByButton
                    value="release date"
                    selected={props.sortBy === 'release date'}
                    handleSortByClick={props.onSortByChange}
                />

                <SortByButton
                    value="rating"
                    selected={props.sortBy === 'rating'}
                    handleSortByClick={props.onSortByChange}
                />
            </div>
        </div>
    </div>
));
