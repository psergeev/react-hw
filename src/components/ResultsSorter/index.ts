import ResultsSorter from './ResultsSorter';

export default ResultsSorter;
