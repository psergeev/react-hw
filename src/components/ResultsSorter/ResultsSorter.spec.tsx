import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ResultsSorter from './ResultsSorter';

jest.mock('../SortByButton', () => 'SortByButton');

const props = {
    moviesCount: 10,
    sortBy: 'rating',
    onSortByChange: jest.fn(),
};

test('ResultsSorter component works as expected', () => {
    const component = renderer.create(
        <ResultsSorter
            {...props}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('ResultsSorter should switch sort by button', () => {
    const testProps = { ...props, sortBy: 'release date' };
    const component = shallow(<ResultsSorter {...testProps} />);

    expect(component).toMatchSnapshot();
});
