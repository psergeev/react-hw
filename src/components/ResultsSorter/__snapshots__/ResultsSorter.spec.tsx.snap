// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSorter component works as expected 1`] = `
<div
  className="results-sorter"
>
  <div
    className=""
  >
    <div>
      10 movie(s) found
    </div>
    <div>
      <span>
        Sort by
      </span>
      <SortByButton
        handleSortByClick={[MockFunction]}
        selected={false}
        value="release date"
      />
      <SortByButton
        handleSortByClick={[MockFunction]}
        selected={true}
        value="rating"
      />
    </div>
  </div>
</div>
`;

exports[`ResultsSorter should switch sort by button 1`] = `
<div
  className="results-sorter"
>
  <div
    className=""
  >
    <div>
      10 movie(s) found
    </div>
    <div>
      <span>
        Sort by
      </span>
      <SortByButton
        handleSortByClick={[MockFunction]}
        selected={true}
        value="release date"
      />
      <SortByButton
        handleSortByClick={[MockFunction]}
        selected={false}
        value="rating"
      />
    </div>
  </div>
</div>
`;
